src/DownloadPage.tsx

```diff
@@ -0,0 +1,184 @@
+import { DownloadIcon } from "lucide-react";
+
+type Platform = "windows" | "mac" | "linux" | "ios" | "android" | "unknown";
+
+const PlatformMetadata: Record<
+  Platform,
+  { name: string; icon: IconDefinition }
+> = {
+  windows: {
+    name: "Windows",
+    icon: faWindows,
+  },
+  mac: {
+    name: "macOS",
+    icon: faApple,
+  },
+  linux: {
+    name: "Linux",
+    icon: faLinux,
+  },
+  ios: {
+    name: "iOS",
+    icon: faApple,
+  },
+  android: {
+    name: "Android",
+    icon: faAndroid,
+  },
+  unknown: {
+    name: "Unknown",
+    icon: faAndroid,
+  },
+};
+
+type DownloadLink = {
+  name: string;
+  urlPattern: RegExp;
+};
+
+const DownloadLinks: Record<string, DownloadLink> = {
+  windows_exe: {
+    name: "Windows (exe)",
+    urlPattern: /.*.exe/,
+  },
+  windows_msi: {
+    name: "Windows (msi)",
+    urlPattern: /.*.msi/,
+  },
+  macos: {
+    name: "macOS",
+    urlPattern: /.*.dmg/,
+  },
+  linux_appimage: {
+    name: "Linux (AppImage)",
+    urlPattern: /.*.AppImage/,
+  },
+  linux_deb: {
+    name: "Linux (deb)",
+    urlPattern: /.*.deb/,
+  },
+};
+
+const PlatformLinks: Record<Platform, DownloadLink[]> = {
+  windows: [DownloadLinks.windows_exe, DownloadLinks.windows_msi],
+  mac: [DownloadLinks.macos],
+  linux: [DownloadLinks.linux_appimage, DownloadLinks.linux_deb],
+  ios: [],
+  android: [],
+  unknown: [],
+};
+
+function detectPlatform(): Platform {
+  if (typeof window === "undefined") return "unknown";
+
+  const userAgent = window.navigator.userAgent.toLowerCase();
+
+  if (userAgent.indexOf("win") > -1) return "windows";
+  if (userAgent.indexOf("mac") > -1) return "mac";
+  if (userAgent.indexOf("linux") > -1) return "linux";
+  if (/iphone|ipad|ipod/.test(userAgent)) return "ios";
+  if (userAgent.indexOf("android") > -1) return "android";
+
+  return "unknown";
+}
+
+function getUrlFromPattern(assets: string[], pattern: RegExp) {
+  const asset = assets.find((asset) => pattern.test(asset));
+  return asset;
+}
+
+export const Download = () => {
+  const [platform, setPlatform] = useState<Platform>("unknown");
+  const [showAll, setShowAll] = useState(false);
+  const [releaseAssets, setReleaseAssets] = useState([]);
+  const [version, setVersion] = useState("");
+
+  useEffect(() => {
+    const platform = detectPlatform();
+    setPlatform(platform);
+    if (PlatformLinks[platform].length === 0) {
+      setShowAll(true);
+    }
+
+    fetch("https://api.github.com/repos/zmkfirmware/zmk-studio/releases/latest")
+      .then((response) => response.json())
+      .then((data) => {
+        setReleaseAssets(
+          data.assets.map((asset: any) => asset.browser_download_url),
+        );
+        setVersion(data.tag_name);
+      });
+  }, []);
+
+  return (
+    <div className="bg-base-200 dark:bg-base-300 text-base-content min-h-full w-full flex flex-col justify-center items-center p-10 pb-48">
+      <img src="/zmk-mac-download.webp" alt="ZMK Studio" className="w-64" />
```

Comment:
What's with the filename here?

Reply:
It's a capture of the mac app icon (added in #104) for the download page hence the name. I can make it more generic, but wanted it to be clear that it's using the mac app's icon.

Reply:
I think I'd rather name it something like `zmk-mac-app-icon.webp` in that case. Thanks.